geopandas/io/sql.py

```diff
@@ -61,36 +64,47 @@ def _df_to_geodf(df, geom_col="geom", crs=None):
     GeoDataFrame
     """
 
-    if geom_col not in df:
-        raise ValueError("Query missing geometry column '{}'".format(geom_col))
+    # sanitize geom_col to always be a list of strings
+    geom_cols = [geom_col] if isinstance(geom_col, str) else geom_col
 
-    geoms = df[geom_col].dropna()
+    for geom_col in geom_cols:
+        if geom_col not in df:
+            raise ValueError("Query missing geometry column '{}'".format(geom_col))
 
-    if not geoms.empty:
-        load_geom_bytes = shapely.wkb.loads
-        """Load from Python 3 binary."""
+    geoms_from_cols = [df[geom_col].dropna() for geom_col in geom_cols]
 
-        def load_geom_buffer(x):
-            """Load from Python 2 binary."""
-            return shapely.wkb.loads(str(x))
+    gdf_list = [df[df.columns.difference(geom_cols)]]
+    for geom_col, geoms in zip(geom_cols, geoms_from_cols):
+        if not geoms.empty:
+            load_geom_bytes = shapely.wkb.loads
+            """Load from Python 3 binary."""
 
-        def load_geom_text(x):
-            """Load from binary encoded as text."""
-            return shapely.wkb.loads(str(x), hex=True)
+            def load_geom_buffer(x):
+                """Load from Python 2 binary."""
+                return shapely.wkb.loads(str(x))
 
-        if isinstance(geoms.iat[0], bytes):
-            load_geom = load_geom_bytes
-        else:
-            load_geom = load_geom_text
+            def load_geom_text(x):
+                """Load from binary encoded as text."""
+                return shapely.wkb.loads(str(x), hex=True)
+
+            if isinstance(geoms.iat[0], bytes):
+                load_geom = load_geom_bytes
+            else:
+                load_geom = load_geom_text
 
-        df[geom_col] = geoms = geoms.apply(load_geom)
-        if crs is None:
-            srid = shapely.geos.lgeos.GEOSGetSRID(geoms.iat[0]._geom)
-            # if no defined SRID in geodatabase, returns SRID of 0
-            if srid != 0:
-                crs = "epsg:{}".format(srid)
+            df[geom_col] = geoms = geoms.apply(load_geom)
+            if crs is None:
+                srid = shapely.geos.lgeos.GEOSGetSRID(geoms.iat[0]._geom)
+                # if no defined SRID in geodatabase, returns SRID of 0
+                if srid != 0:
+                    crs = "epsg:{}".format(srid)
 
-    return GeoDataFrame(df, crs=crs, geometry=geom_col)
+        gdf_list.append(GeoDataFrame(df[[geom_col]], crs=crs, geometry=geom_col))
+
+    # set first geom_col as active geometry
+    return GeoDataFrame(
+        pd.concat(gdf_list, axis=1)[df.columns], crs=crs, geometry=geom_cols[0]
```

Comment:
Can we skip concat if there's only one column and return `gdf_list[0]` directly?